File: tests/src/com.fasterxml.jackson.module/jackson-module-kotlin/2.22.0/src/test/java/com_fasterxml_jackson_module/jackson_module_kotlin/ReflectionCacheKtTest.java
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package com_fasterxml_jackson_module.jackson_module_kotlin;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.json.JsonMapper;
import com.fasterxml.jackson.module.kotlin.KotlinModule;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class ReflectionCacheKtTest {
    @Test
    void deserializesPrimaryConstructorWithValueClassParameter() throws Exception {
        ObjectMapper objectMapper = newObjectMapper();

        ReflectionCacheValueClassHolder result = objectMapper.readValue(
                "{\"wrapped\":\"alpha\"}",
                ReflectionCacheValueClassHolder.class);

        assertThat(result.wrappedValue()).isEqualTo("alpha");
    }

    private static ObjectMapper newObjectMapper() {
        return JsonMapper.builder()
                .addModule(new KotlinModule.Builder().build())
                .build();
    }
}

File: tests/src/com.fasterxml.jackson.module/jackson-module-kotlin/2.22.0/src/test/kotlin/com_fasterxml_jackson_module/jackson_module_kotlin/ReflectionCacheKtTest.kt
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package com_fasterxml_jackson_module.jackson_module_kotlin

import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

public class ReflectionCacheKtTest {
    @Test
    fun deserializesPrimaryConstructorWithValueClassParameter() {
        val objectMapper: ObjectMapper = newObjectMapper()

        val result: ReflectionCacheValueClassHolder = objectMapper.readValue(
            """{"wrapped":"alpha"}""",
            ReflectionCacheValueClassHolder::class.java
        )

        assertThat(result.wrappedValue()).isEqualTo("alpha")
    }

    private fun newObjectMapper(): ObjectMapper = jacksonObjectMapper()
}
